### PR TITLE
Rename stdlib_stats_distribution_PRNG to stdlib_random

### LINKED ISSUE #544: Collect distribution modules in stdlib_stats_distribution
@Jim-215-Fisher in https://github.com/fortran-lang/stdlib/pull/538#issuecomment-932976436

> Regarding to stdlib_stats_distribution_PRNG module, I am not sure whether it should be in stats_distribution package. GSL has a separate random numbers package in addition to random distributions and statistics.

Should `stdlib_stats_distribution_PRNG` become a submodule of `stdlib_stats_distribution`?

### PR DESCRIPTION
**Pull request by awvwgk**
Closes #544

### CODE REVIEW

**Review by milancurcic — APPROVED** [review]

**Review by gareth-nx — APPROVED** [review]
Looks good, thanks

File: src/stdlib_random.fypp [hunk 1]
```diff
@@ -18,7 +18,7 @@ module stdlib_stats_distribution_PRNG
     !! Version experimental
     !!
     !! Generation of random integers with different kinds
-    !! ([Specification](../page/specs/stdlib_stats_distribution_PRNG.html#
+    !! ([Specification](../page/specs/stdlib_random.html#
     !! description))
```
Comment (jvdp1) [inline src/stdlib_random.fypp:22]: While modiffyng this module, is it also ok to update the link to a more explicit one (instead of desctiption)?

**Review by jvdp1 — APPROVED** [review]
Thank you.

File: src/stdlib_random.fypp (hunk 1 above)
Comment (awvwgk) [reply src/stdlib_random.fypp]: Done.

**Pull request merged**